src/main/java/org/openrewrite/maven/ResourceParser.java

```diff
@@ -106,4 +111,55 @@ public <S extends SourceFile> List<S> parseSourceFiles(
             throw new UncheckedIOException(e);
         }
     }
+
+    public <S extends SourceFile> List<S> parseQuarks(
```

Comment:
I believe there is a static method on the QuarkParser that you can use rather than rolling your own. 

The only issue here is that the static method does not have the ability to pass a separate `baseDir` and `searchDir` and it does not allow an ExecutionContext to be passed to it. 

I would just overload the static method in QuarkParser to add these two parameters. 

This method becomes much simplier: 

```
List<Quark> quarks = QuarkParser.parseAllOtherFiles(baseDir, searchDir, alreadyParsed, ctx);
alreadyParsed.addAll(quarks.stream.map(SourceFile::getPath).collect(Collectors.toSet()));
return (List<S>) quarks;
```